Lib/test/test_profile.py

```diff
@@ -131,6 +132,116 @@ def test_output_file_when_changing_directory(self):
+        cmd = [sys.executable, '-m', 'profile'] + list(args)
+        proc = subprocess.Popen(
+            cmd,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            universal_newlines=True
+        )
+        stdout, stderr = proc.communicate()
+        return proc.returncode, stdout, stderr
+
+    def test_basic_profile(self):
+        """Test basic profiling of a script."""
+        returncode, stdout, stderr = self._run_profile_cli(self.script_file)
+        self.assertEqual(returncode, 0)
+        self.assertIn("function calls", stdout)
+        self.assertIn("factorial", stdout)
+
+    def test_sort_options(self):
+        """Test different sort options."""
+        # List of sort options known to work
+        sort_options = ['calls', 'cumulative', 'cumtime', 'file',
+                        'filename', 'module', 'ncalls', 'pcalls',
+                        'line', 'stdname', 'time', 'tottime']
+
+        # Test each sort option individually
+        for option in sort_options:
+            with self.subTest(sort_option=option):
+                returncode, stdout, stderr = self._run_profile_cli(
+                    '-s', option, self.script_file
+                )
+                self.assertEqual(returncode, 0)
+                self.assertIn("function calls", stdout)
+
+    def test_output_file(self):
+        """Test writing profile results to a file."""
+        output_file = TESTFN + '.prof'
+        self.addCleanup(unlink, output_file)
+
+        returncode, stdout, stderr = self._run_profile_cli(
+            '-o', output_file, self.script_file
+        )
+        self.assertEqual(returncode, 0)
+
+        # Check that the output file exists and contains profile data
+        self.assertTrue(os.path.exists(output_file))
+        stats = pstats.Stats(output_file)
+        self.assertGreater(stats.total_calls, 0)
+
+    def test_invalid_option(self):
+        """Test behavior with an invalid option."""
+        returncode, stdout, stderr = self._run_profile_cli(
+            '--invalid-option', self.script_file
+        )
+        self.assertNotEqual(returncode, 0)
+        self.assertIn("error", stderr.lower())
+
+    def test_no_arguments(self):
+        """Test behavior with no arguments."""
+        returncode, stdout, stderr = self._run_profile_cli()
+        self.assertNotEqual(returncode, 0)
+
+        # Check either stdout or stderr for usage information
+        combined_output = stdout.lower() + stderr.lower()
+        self.assertTrue(
+            "usage:" in combined_output or
+            "error:" in combined_output or
+            "no script filename specified" in combined_output,
+            "Expected usage information or error message not found"
+        )
+
+    def test_run_module(self):
+        """Test profiling a module with -m option."""
+        # Create a small module
+        module_name = "test_profile_module"
+        module_file = f"{module_name}.py"
```

Comment:
Why?
```suggestion
        module = f"profile_module.py"
```
It can be simple